validation-test/stdlib/String.swift

```diff
@@ -2370,4 +2370,23 @@ StringTests.test("SmallString.zeroTrailingBytes") {
   }
 }
 
+StringTests.test("String.CoW.reserveCapacity") {
+  for _ in 0..<100 {
+    var str = String(repeating: "a", count: 100)
+    var copy = str
+    copy.reserveCapacity(20)
+    copy.append(contentsOf: String(repeating: "z", count: 50))
+  }
```

Comment:
It would be nice to document why these two tests exist, and/or to add some actual checks to them. (For example, we could check that `copy` has the expected contents at the end.)

Why does this test has a loop?